shared/subscribers/stream_subscriber.go

```diff
@@ -80,7 +94,7 @@ func getBufSize(batch int, types []events.Type) int {
 	return base
 }
 
-func NewStreamSub(ctx context.Context, types []events.Type, batchSize int, filters ...EventFilter) *StreamSub {
+func NewStreamSub(ctx context.Context, types []events.Type, batchSize int, filters ...EventFilter) StreamSubscription {
```

Comment:
This is considered bad practice in golang. A function should be as liberal as possible in terms of its returns, but as restrictive as possible in terms of what it accepts as arguments. Returning the underlying type as in interface defined in this package itself is doing the opposite

Reply:
it was due needing the internal context of the event_observer for the test (need to bend the black box testing rules cause of the threading) , which ultimately required this change  however just realised the context is not needed can just rely on the out channel closing, this has been moved back to *StreamSub